### ISSUE TITLE
seahorse: couldn't add keyring - no such secret collection at path: /

### ISSUE THREAD

**Issue by proteansec** [NixOS/nixpkgs#14966]
When installing and starting seahorse, it will display the following error, which includes the warning about saving all application settings in memory only, since GSettings does not work properly.

> seahorse 
> GLib-GIO-Message: Using the 'memory' GSettings backend.  Your settings will not be saved or shared with other applications.
> Xlib:  extension "GLX" missing on display ":0.0".
> *\* (seahorse:17980): WARNING **: Error retrieving accessibility bus address: org.freedesktop.DBus.Error.ServiceUnknown: The name org.a11y.Bus was not provided by any .service files
> seahorse-Message: DNS-SD initialization failed: Daemon not running
> seahorse-Message: unsupported key server uri scheme: ldap

To solve the GSettings warning, I've set the `--prefix GIO_EXTRA_MODULES : "${gnome3.dconf}/lib/gio/modules"` as documented at #5433. After restarting seahorse it will no longer complain about the GSettings.

> /nix/store/j1967vbm6zisc09lf0bd44wz6dqs5sby-seahorse-3.18.0/bin/seahorse
> Xlib:  extension "GLX" missing on display ":0.0".
> *\* (seahorse:17481): WARNING *_: Error retrieving accessibility bus address: org.freedesktop.DBus.Error.ServiceUnknown: The name org.a11y.Bus was not provided by any .service files
> seahorse-Message: DNS-SD initialization failed: Daemon not running
> seahorse-Message: unsupported key server uri scheme: ldap
> (seahorse:17481): dconf-WARNING *_: failed to commit changes to dconf: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name ca.desrt.dconf was not provided by any .service files

The problem is that after starting seahorse, there is no default keyring. We can try to create new password keyring by choosing the 'Password Keyring' option as presented below.

![seahorse1](https://cloud.githubusercontent.com/assets/1753498/14776507/4905c3de-0ac7-11e6-9b16-46ec0dd44e2d.png)

However adding a new password keyring results in the following error.

![seahorse2](https://cloud.githubusercontent.com/assets/1753498/14776529/60f8afba-0ac7-11e6-8f32-64da16ecf938.png)

It's normally the case that password keyring is created under ~/.gnome2/keyrings, which is a folder that is currently created and has valid permissions.

> ls -ld ~/.gnome2/keyrings
> drwx------ 2 user user 6 Apr 24 23:49 /home/user/.gnome2/keyrings

We should try to determine why the default keyring is not created, but even if it isn't created by default, we can always create it manually. More importantly, we have to discover why we're unable to create a new password keyring.

**Comment by spinus** [comment]
I can confirm the error, have the same. Btw, the path for keyring is ~/.local/share/keyrings.

**Comment by dimti** [comment]
@proteansec Hi. From your description - exists solve this problem or no?

I confirm similar problem. Unable to add new keyring...

**Comment by proteansec** [comment]
Hi,

I have the following in my configuration.nix, but the problem is not resolved:

`
services = {
    gnome3 = {
      gnome-keyring.enable = true;
      seahorse.enable = true;
      at-spi2-core.enable = true;
    };
    dbus.packages = [ pkgs.gnome3.gnome_keyring pkgs.gnome3.gcr pkgs.gnome.GConf ];
  };

`

**Comment by proteansec** [comment]
Just today I've restarted a computer due to the system freezing - not sure what happened though. But after a restart, everything suddenly started to work. Now I'm able to create a new keyring in seahorse without any problems. My configuration.nix is the following:

```
  environment.systemPackages = with pkgs; [
    gnome3.gnome_keyring
    python27Packages.keyring
  }

  services = {
    gnome3 = {
      gnome-keyring.enable = true;
      seahorse.enable = true;
      at-spi2-core.enable = true;
    };
    dbus.packages = [ pkgs.gnome3.gnome_keyring pkgs.gnome3.gcr pkgs.gnome.GConf ];
  };
```

Also, remember to restart the computer in order for changes to take effect. Alternatively, you can investigate the exact service that needs to be restarted and do so manually, or the change needs to be incorporated into the nix when applying the new configuration with **nixos-rebuild**.

Can somebody test the changes to confirm whether this solves the problem or not.

**Comment by spinus** [comment]
hah :-) very good man. Thank you for sharing.

I'll try your config and report back.

**Comment by oleks** [comment]
This fixed the issue for me.

**Comment by spinus** [comment]
Forgot :( I can confirm as well.

**Comment by dustindorroh** [comment]
I just had to remake my key ring directory
```
mkdir ~/.local/share/keyrings
```

**Comment by alanpearce** [comment]
I also needed to add
```
programs.dconf.enable = true;
```
and remove
```
services.dbus.socketActivated = true;
```

**Comment by sarpik** [comment]
For anyone still having this issue:
I've found a fix @ https://unix.stackexchange.com/a/295652 , originally from https://bbs.archlinux.org/viewtopic.php?id=231006

TL;DR:

`~/.xinitrc`

```sh
mkdir -p "$HOME"/.local/share/keyrings
source /etc/X11/xinit/xinitrc.d/50-systemd-user.sh
```

Re-login might be required (or just `source /etc/X11/xinit/xinitrc.d/50-systemd-user.sh` and then re-launch `seahorse`)

**Comment by jtojnar** [comment]
The original issue was caused by missing dconf GIO module and was fixed long time ago. Please open a new issue if you still encounter some problems.
